includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -922,6 +923,15 @@ private function validate_cdata_for_node( DOMElement $element, $cdata_spec ) {
 				return [ 'code' => self::MANDATORY_CDATA_MISSING_OR_INCORRECT ];
 			}
 		}
+
+		// When the CDATA is expected to be JSON, ensure it's valid JSON.
+		if ( 'script' === $element->tagName &&
+			in_array( $element->getAttribute( 'type' ), [ 'application/json', 'application/ld+json' ], true ) &&
+			null === json_decode( $element->textContent )
+		) {
+			return [ 'code' => self::INVALID_JSON_CDATA ];
```

Comment:
This could include the JSON error code as part of the error data.

Reply:
Here's the new error message added in https://github.com/ampproject/amp-wp/pull/4340/commits/ea26570a49ca6f3cbfc8a32c7e64aab3a4a91803. How does that look?

![new-error-message](https://user-images.githubusercontent.com/4063887/75597027-c9f32b00-5a58-11ea-8d6c-e1fb964e7d95.png)

Reply:
The most common error would probably be probably 'Syntax error'.